# HG changeset patch
# User Rik <rik@octave.org>
# Date 1414427446 25200
#      Mon Oct 27 09:30:46 2014 -0700
# Branch stable
# Node ID f96495e88a70e54694830b5c5f3b01f79c5beecc
# Parent  eabfde4b6c5679431a6a5f32f5271697fc3d1de4
Allow setting horizontal vector of "ydata" in bar plots (bug #43468).

* __bar__.m (update_data): Check for vector of ydata and always transform
to column vector.

diff --git a/scripts/plot/draw/private/__bar__.m b/scripts/plot/draw/private/__bar__.m
--- a/scripts/plot/draw/private/__bar__.m
+++ b/scripts/plot/draw/private/__bar__.m
@@ -432,16 +432,18 @@ function update_data (h, ~)
       hlist = get (h, "bargroup");
       x = get (h, "xdata");
       if (! isvector (x))
         x = x(:);
       endif
       ydat = get (hlist, "ydata");
       if (iscell (ydat))
         y = cell2mat (ydat.');
+      elseif (isvector (ydat))
+        y = ydat(:);
       else
         y = ydat;
       endif
 
       [xb, yb] = bar (x, y, get (h, "barwidth"), get (h, "barlayout"),
                       "basevalue", get (h, "basevalue"));
 
       vertical = strcmp (get (h, "horizontal"), "off");
